src/Libraries/DSOffice/OpenXmlHelper.cs

```diff
@@ -0,0 +1,647 @@
+                    if (cell.CellValue.TryGetBoolean(out var value))
+                    {
+                        if (readAsString)
+                        {
+                            return value.ToString();
+                        }
+
+                        return value;
+                    }
+
+                    return cell.InnerText;
+                }
+                else
+                {
+                    // Default to raw string value for inline strings and unknown types.
+                    return cell.InnerText;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the worksheet with the specified name from the document.
+        /// </summary>
+        /// <param name="document">Document where to obtain the worksheet from</param>
+        /// <param name="sheetName">Name of the sheet to obtain</param>
+        /// <returns>The sheet with the specified name or null if it was not found</returns>
+        private static WorksheetPart GetWorksheetPart(SpreadsheetDocument document, string sheetName)
+        {
+            var sheet = document.WorkbookPart.Workbook.GetFirstChild<Sheets>().Elements<Sheet>().FirstOrDefault(s => s.Name == sheetName);
+            if (sheet == null)
+            {
+                return null;
+            }
+            return document.WorkbookPart.GetPartById(sheet.Id.Value) as WorksheetPart;
+        }
+
+        /// <summary>
+        /// Checks whether the passed in data contains a function.
+        /// </summary>
+        /// <param name="data"></param>
+        /// <returns></returns>
+        private static bool ContainsFunction(object[][] data)
+        {
+            foreach (var row in data)
+            {
+                if (row == null)
+                {
+                    continue;
+                }
+
+                foreach (var value in row)
+                {
+                    if (value is ProtoCore.DSASM.StackValue stackValue && stackValue.IsPointer)
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Opens an existing workbook or creates one from scratch on the specified file path.
+        /// </summary>
+        /// <param name="filePath">File path where the workbook is or should be created</param>
+        /// <param name="sheetName">In case a new workbook is created, the sheet will use this name</param>
+        /// <returns>A spreadsheet document</returns>
+        private static SpreadsheetDocument OpenOrCreate(string filePath, string sheetName)
+        {
+            if (File.Exists(filePath))
+            {
+                return SpreadsheetDocument.Open(filePath, true);
+            }
+            else
+            {
+                var document = SpreadsheetDocument.Create(filePath, SpreadsheetDocumentType.Workbook);
```

Comment:
Curious if this function would be interrupted with exception that user only have read but not write access to the target path

Reply:
I think either of the code paths of this function should fail in that case, because the above path opens the file to write it.

Reply:
@mmisol OK. I am not sure if you have a local env to test this. I can make a note to QA this case if you want?

Reply:
I will give this a try but yes, it would be a good thing to try by part of the QA team.

Reply:
The message I get is `Access to the path '<PATH>' is denied.` which must come from Windows.

Reply:
@mmisol Thanks, as long as Dynamo does not crash and we display this as a warning or so. Appreciate the checking!